plugins/kubernetes/app/views/admin/kubernetes/clusters/_capacity.html.erb

```diff
@@ -0,0 +1,33 @@
+<%# avoiding N+1 for every cluster by doing all at once ... ideally do 1 query to get both values %>
+<%
+  @used_cpu ||= Kubernetes::DeployGroupRole.group(:deploy_group_id).sum("requests_cpu * replicas")
+  @used_memory ||= Kubernetes::DeployGroupRole.group(:deploy_group_id).sum("requests_memory * replicas")
+  nodes = cluster.client.get_nodes rescue []
+  available_cpu = nodes.sum { |n| n.dig(:status, :capacity, :cpu).to_i }
+  available_memory = nodes.sum { |n| n.dig(:status, :capacity, :memory).to_i / 1024 / 1024 }
+  used_cpu = cluster.deploy_groups.sum { |dg| @used_cpu[dg.id] || 0 }
+  used_memory = cluster.deploy_groups.sum { |dg| @used_memory[dg.id] || 0 } / 1024
+  used_cpu_percent = (available_cpu.zero? ? 0 : (used_cpu * 100.0 / available_cpu).round)
+  used_memory_percent = (available_memory.zero? ? 0 : (used_memory * 100.0 / available_memory).round)
```

Comment:
Seems like enough logic here that it would be best putting it into model code, rather than a view.